### PR TITLE
Add sorting to the project manager

### PR DESCRIPTION
**Pull request by helenaMontenegro**
We added to option to order the projects in the Project Manager in alphabetic order, here's how it looks:

byname

This solves issue [#8794](https://github.com/godotengine/godot/issues/8794)

### CODE REVIEW

**Comment by groud** [comment]
I think adding another line for that takes too much place. The sorting option should be on the same line as the search input.

**Comment by akien-mga** [comment]
I agree that it should be on the same line.

There are also style issues that would need to be fixed, see the diff in https://travis-ci.org/godotengine/godot/jobs/458927443 and [Code style guidelines](http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html).

Please also amend the commit log (with `git commit --amend`, see [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html)) to make it more explicit. It's good that the log mentions the issue number, but it should also explain in plain words what this issue (or more specifically the commit's change) is about.

Something like:
```
Add sorting to the project manager

Fixes #8794.
```
would be much more readable in `git log`.

File: editor/project_manager.cpp [hunk 1]
```diff
@@ -875,9 +876,16 @@ struct ProjectItem {
...
 	}
-	_FORCE_INLINE_ bool operator<(const ProjectItem &l) const { return last_modified > l.last_modified; }
+	_FORCE_INLINE_ bool operator<(const ProjectItem &l) const { 
+		if(orderedLatestModification)
+			return last_modified > l.last_modified; 
+		return project < l.project;
+	}
 	_FORCE_INLINE_ bool operator==(const ProjectItem &l) const { return project == l.project; }
+	public:
+	static void setOrderedLatestModification(bool value) { orderedLatestModification = value; }
```
Comment (akien-mga) [inline editor/project_manager.cpp]: As you can see in surrounding code, method and variable identifiers in Godot's code base use `snake_case`, so you should do the same for new code.

File: editor/project_manager.cpp [hunk 2]
```diff
@@ -1818,12 +1832,35 @@ ProjectManager::ProjectManager() {
...
 	VBoxContainer *search_tree_vb = memnew(VBoxContainer);
-	search_tree_vb->set_h_size_flags(SIZE_EXPAND_FILL);
 	tree_hb->add_child(search_tree_vb);
 
+	HBoxContainer *sort_box = memnew(HBoxContainer);
+	Label *sort_label = memnew(Label);
+	sort_label->set_text(TTR("Sort:"));
+	sort_box->add_child(sort_label);
+	sort_box->add_spacer(true);
+	Vector<String> vec1;
```
Comment (akien-mga) [inline editor/project_manager.cpp]: Please give the variable a clearer name. `filters` seems like a natural fit, or `filter_names`.

Comment (akien-mga) [reply editor/project_manager.cpp]: I see there's another one later on, so maybe `sort_filters` here and `search_filters` for the other.

**Comment by helenaMontenegro** [comment]
Thank you so much for the feedback!

This is how it looks now:
bynamelatest

I added a label for the search filter since I think the respective drop-down menu wasn't very explicit, but I can take it out. What do you think?

**Comment by asheraryam** [comment]
> 
> 
> Thank you so much for the feedback!
> 
> This is how it looks now:
> bynamelatest
> 
> I added a label for the search filter since I think the respective drop-down menu wasn't very explicit, but I can take it out. What do you think?

I think the label is fine, but the text should say "Last Modified" instead of "Latest Modification". It's more commonly used and takes less horizontal space.

And thanks! Awesome PR.

**Comment by helenaMontenegro** [comment]
"Last Modified" is way better than "Last Modification"! Thank you! 

Now it looks like this:
howitlooksnow

**Comment by helenaMontenegro** [comment]
I'm so sorry there's so many commits, I was having trouble working with git bash.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**